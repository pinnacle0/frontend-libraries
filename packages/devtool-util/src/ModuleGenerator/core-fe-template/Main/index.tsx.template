import React from "react";
import {ReactUtil} from "@pinnacle0/web-ui/util/ReactUtil";
import "./index.less";

export const Main = ReactUtil.memo("Main", () => {
    return <div />;
});
