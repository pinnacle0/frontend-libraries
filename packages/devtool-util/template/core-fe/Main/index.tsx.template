import React from "react";
import {{1}Module} from "../module"
import "./index.less";

export const Main: React.ComponentType = {1}Module.attachLifecycle(() => {
    return <div />;
});
