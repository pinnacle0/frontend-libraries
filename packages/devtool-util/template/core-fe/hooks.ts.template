import {useSelector} from "core-fe";
import type {RootState} from "type/state";

export function use{1}State<T>(fn: (state: RootState["app"]["{2}"]) => T): T {
    return useSelector((state: RootState) => fn(state.app.{2}));
}
