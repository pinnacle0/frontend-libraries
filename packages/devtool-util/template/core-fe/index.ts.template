export {Main} from "./Main";
export {actions} from "./module";

