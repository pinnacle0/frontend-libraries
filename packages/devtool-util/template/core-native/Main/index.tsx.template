import React from "react";
import {StyleSheet, View} from "react-native";
import {ReactUtil} from "@pinnacle0/web-ui/util/ReactUtil";

export const Main = ReactUtil.memo("Main", () => {
    return (
        <View style={styles.container}>
            <View />
        </View>
    );
});

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    },
});
