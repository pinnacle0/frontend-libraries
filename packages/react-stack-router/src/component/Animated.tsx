import React, {useEffect, useRef} from "react";

interface Props extends React.DetailedHTMLProps<React.HtmlHTMLAttributes<HTMLDivElement>, HTMLDivElement> {
    ref?: React.MutableRefObject<HTMLDivElement> | React.RefCallback<HTMLDivElement>;
    onEnter: Keyframe[] | PropertyIndexedKeyframes | null;
    onExit: Keyframe[] | PropertyIndexedKeyframes | null;
    duration?: number;
}

export const Animated = (props: Props) => {
    const {
        children,
        onEnter,
        onExit,
        ref,
        duration = 300,
        __removed,
        __onExited,
        className,
        ...restProps
    } = props as Props & {
        __onExited?: () => void;
        __removed?: boolean;
    };
    const elementRef = useRef<HTMLDivElement | null>(null);
    const compositedRef = ref ? compositeRef(elementRef, ref) : elementRef;

    const onExitRef = useRef(__onExited);
    onExitRef.current = __onExited;

    const animationRef = useRef({
        duration,
        onEnter,
        onExit,
    });

    animationRef.current = {
        duration,
        onEnter,
        onExit,
    };

    useEffect(() => {
        const el = elementRef.current;
        const {onEnter, duration} = animationRef.current;
        if (el) {
            const animation = el.animate(onEnter, {
                duration,
                fill: "forwards",
                easing: "cubic-bezier(.05,.74,.3,1.01)",
            });
            return () => animation.cancel();
        }
    }, []);

    useEffect(() => {
        const el = elementRef.current;
        const {onExit, duration} = animationRef.current;
        if (__removed && el) {
            const animation = el.animate(onExit, {
                duration,
                easing: "cubic-bezier(.05,.74,.3,1.01)",
                fill: "forwards",
            });
            animation.onfinish = () => onExitRef.current?.();
            return () => animation.cancel();
        }
    }, [__removed]);

    return (
        <div {...restProps} className={`g-animated ${className ?? ""}`} ref={compositedRef}>
            {children}
        </div>
    );
};

function compositeRef<T>(...refs: Array<React.RefCallback<T> | React.MutableRefObject<T | null>>) {
    return (node: T | null) => {
        refs.forEach(ref => {
            if (typeof ref === "function") {
                ref(node);
            } else {
                ref.current = node;
            }
        });
    };
}
