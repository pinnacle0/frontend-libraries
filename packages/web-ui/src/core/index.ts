export {type Props as CarouselProps} from "./Carousel";
export {Carousel} from "./Carousel";
export {type Props as YearMonthSelectorProps} from "./YearMonthSelector";
export {YearMonthSelector} from "./YearMonthSelector";
export {type Props as RelativeTimeProps} from "./RelativeTime";
export {RelativeTime} from "./RelativeTime";
export {type Props as WithTooltipListProps} from "./WithTooltipList";
export {WithTooltipList} from "./WithTooltipList";
export {type Props as ButtonProps} from "./Button";
export {Button} from "./Button";
export {type Props as TimeRangePickerProps} from "./TimeRangePicker";
export {TimeRangePicker} from "./TimeRangePicker";
export {type Props as SpinProps} from "./Spin";
export {Spin} from "./Spin";
export {type Props as CardProps} from "./Card";
export {Card} from "./Card";
export {type Props as RadioProps} from "./Radio";
export {Radio} from "./Radio";
export {type Props as SkeletonProps} from "./Skeleton";
export {Skeleton} from "./Skeleton";
export {type Props as AmountRangeInputProps} from "./AmountRangeInput";
export {AmountRangeInput} from "./AmountRangeInput";
export {type Props as AutoCompleteInputProps} from "./AutoCompleteInput";
export {AutoCompleteInput} from "./AutoCompleteInput";
export {type Props as OrientationContextProps} from "./OrientationContext";
export {OrientationContextProvider} from "./OrientationContext";
export {type Props as BadgeProps} from "./Badge";
export {Badge} from "./Badge";
export {type Props as DateRangePickerProps} from "./DateRangePicker";
export {DateRangePicker} from "./DateRangePicker";
export {type Props as BoolSwitchProps} from "./BoolSwitch";
export {BoolSwitch} from "./BoolSwitch";
export {type Props as LabelledSelectProps} from "./LabelledSelect";
export {LabelledSelect} from "./LabelledSelect";
export {Progress} from "./Progress";
export {type Props as TooltipProps} from "./Tooltip";
export {Tooltip} from "./Tooltip";
export {type Props as SignedAmountInputProps} from "./SignedAmountInput";
export {SignedAmountInput} from "./SignedAmountInput";
export {type Props as DocumentTitleProps} from "./DocumentTitle";
export {DocumentTitle} from "./DocumentTitle";
export {Grid} from "./Grid";
export {EnumSelect} from "./EnumSelect";
export {type Props as EnumSelectProps} from "./EnumSelect";
export {type Props as BoolRadioProps} from "./BoolRadio";
export {BoolRadio} from "./BoolRadio";
export {type Props as BoolSelectProps} from "./BoolSelect";
export {BoolSelect} from "./BoolSelect";
export {type Props as InputProps} from "./Input";
export {Input} from "./Input";
export {type Props as ImageProps} from "./Image";
export {Image} from "./Image";
export {type Props as MultipleEnumSelectProps} from "./MultipleEnumSelect";
export {MultipleEnumSelect} from "./MultipleEnumSelect";
export {type TableProps} from "./Table";
export {Table} from "./Table";
export {type Props as UploaderProps} from "./Uploader";
export {Uploader} from "./Uploader";
export {type Props as LocalImporterProps} from "./LocalImporter";
export {LocalImporter} from "./LocalImporter";
export {type Props as TagsProps} from "./Tags";
export {Tags} from "./Tags";
export {type Props as LinkProps} from "./Link";
export {Link} from "./Link";
export {type Props as TimePickerProps} from "./TimePicker";
export {TimePicker} from "./TimePicker";
export {type Props as SelectProps} from "./Select";
export {Select} from "./Select";
export {type Props as AmountConditionInputProps} from "./AmountConditionInput";
export {AmountConditionInput} from "./AmountConditionInput";
export {type Props as DateCalendarProps} from "./DateCalendar";
export {DateCalendar} from "./DateCalendar";
export {type Props as CheckboxProps} from "./Checkbox";
export {Checkbox} from "./Checkbox";
export {type Props as StepsProps} from "./Steps";
export {Steps} from "./Steps";
export {type Props as DateTimePickerProps} from "./DateTimePicker";
export {DateTimePicker} from "./DateTimePicker";
export {type Props as SearchableSelectProps} from "./SearchableSelect";
export {SearchableSelect} from "./SearchableSelect";
export {type Props as DrawerProps} from "./Drawer";
export {Drawer} from "./Drawer";
export {type Props as TagProps} from "./Tag";
export {Tag} from "./Tag";
export {type Props as DatePickerProps} from "./DatePicker";
export {DatePicker} from "./DatePicker";
export {type Props as LocaleSelectProps} from "./LocaleSelect";
export {LocaleSelect} from "./LocaleSelect";
export {type Props as UIProviderProps} from "./UIProvider";
export {UIProvider} from "./UIProvider";
export {type Props as DropdownProps} from "./Dropdown";
export {Dropdown} from "./Dropdown";
export {type Props as SpaceProps} from "./Space";
export {Space} from "./Space";
export {type Props as CollapseProps} from "./Collapse";
export {Collapse} from "./Collapse";
