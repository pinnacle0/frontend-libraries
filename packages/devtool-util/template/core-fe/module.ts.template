import {Module, register} from "core-fe";
import type {SagaGenerator} from "core-fe";
import type {RootState} from "type/state";
import type {State} from "./type";

const initialState: State = {};

class {1}Module extends Module<RootState, "{2}"> {
    override *onEnter(): SagaGenerator {
        // TODO
    }
}

export const {2}Module = register(new {1}Module("{2}", initialState));
export const actions = {2}Module.getActions();
