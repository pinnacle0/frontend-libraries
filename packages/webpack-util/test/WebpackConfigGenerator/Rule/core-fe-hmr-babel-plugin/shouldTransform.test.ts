import * as babel from "@babel/core";
import plugin from "../../../../src/WebpackConfigGenerator/Rule/core-fe-hmr-babel-plugin";

describe("core-fe-hmr-babel-plugin", () => {
    test("works with `import {Module}`", () => {
        const source = `
            import {Module} from "core-fe";
            import {Main} from "./component/Main";

            const initialState = {};
            class FeatureModule extends Module {}

            const featureModule = register(new FeatureModule("feature", initialState));
            export const actions = featureModule.getActions();
            export const MainComponent = featureModule.attachLifecycle(Main);
        `;
        const {code} = babel.transformSync(source, {plugins: [plugin]})!;
        expect(code).toMatchSnapshot();
        expect(code).toContain("if (module.hot) module.hot.decline();");
    });

    test("works with `import {Module as CoreFEModule}`", () => {
        const source = `
            import {Module as CoreFEModule} from "core-fe";
            import {Main} from "./component/Main";

            const initialState = {};
            class FeatureModule extends CoreFEModule {}

            const featureModule = register(new FeatureModule("feature", initialState));
            export const actions = featureModule.getActions();
            export const MainComponent = featureModule.attachLifecycle(Main);
        `;
        const {code} = babel.transformSync(source, {plugins: [plugin]})!;
        expect(code).toMatchSnapshot();
        expect(code).toContain("if (module.hot) module.hot.decline();");
    });

    test("works with `import CoreFEDefault, {Module}`", () => {
        const source = `
            import CoreFEDefault, {Module} from "core-fe";
            import {Main} from "./component/Main";

            const initialState = {};
            class FeatureModule extends Module {}

            const featureModule = register(new FeatureModule("feature", initialState));
            export const actions = featureModule.getActions();
            export const MainComponent = featureModule.attachLifecycle(Main);
        `;
        const {code} = babel.transformSync(source, {plugins: [plugin]})!;
        expect(code).toMatchSnapshot();
        expect(code).toContain("if (module.hot) module.hot.decline();");
    });

    test("works with `import * as CoreFENamespace`", () => {
        const source = `
            import * as CoreFENamespace from "core-fe";
            import {Main} from "./component/Main";

            const initialState = {};
            class FeatureModule extends CoreFENamespace.Module {}

            const featureModule = register(new FeatureModule("feature", initialState));
            export const actions = featureModule.getActions();
            export const MainComponent = featureModule.attachLifecycle(Main);
        `;
        const {code} = babel.transformSync(source, {plugins: [plugin]})!;
        expect(code).toMatchSnapshot();
        expect(code).toContain("if (module.hot) module.hot.decline();");
    });
});
