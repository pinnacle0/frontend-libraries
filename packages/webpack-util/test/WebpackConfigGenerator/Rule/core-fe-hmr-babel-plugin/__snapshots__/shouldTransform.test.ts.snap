// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core-fe-hmr-babel-plugin works with \`import * as CoreFENamespace\` 1`] = `
"import * as CoreFENamespace from \"core-fe\";
import { Main } from \"./component/Main\";
const initialState = {};

class FeatureModule extends CoreFENamespace.Module {}

const featureModule = register(new FeatureModule(\"feature\", initialState));
export const actions = featureModule.getActions();
export const MainComponent = featureModule.attachLifecycle(Main);
if (module.hot) module.hot.decline();"
`;

exports[`core-fe-hmr-babel-plugin works with \`import {Module as CoreFEModule}\` 1`] = `
"import { Module as CoreFEModule } from \"core-fe\";
import { Main } from \"./component/Main\";
const initialState = {};

class FeatureModule extends CoreFEModule {}

const featureModule = register(new FeatureModule(\"feature\", initialState));
export const actions = featureModule.getActions();
export const MainComponent = featureModule.attachLifecycle(Main);
if (module.hot) module.hot.decline();"
`;

exports[`core-fe-hmr-babel-plugin works with \`import {Module}\` 1`] = `
"import { Module } from \"core-fe\";
import { Main } from \"./component/Main\";
const initialState = {};

class FeatureModule extends Module {}

const featureModule = register(new FeatureModule(\"feature\", initialState));
export const actions = featureModule.getActions();
export const MainComponent = featureModule.attachLifecycle(Main);
if (module.hot) module.hot.decline();"
`;

exports[`core-fe-hmr-babel-plugin works with \`import CoreFEDefault, {Module}\` 1`] = `
"import CoreFEDefault, { Module } from \"core-fe\";
import { Main } from \"./component/Main\";
const initialState = {};

class FeatureModule extends Module {}

const featureModule = register(new FeatureModule(\"feature\", initialState));
export const actions = featureModule.getActions();
export const MainComponent = featureModule.attachLifecycle(Main);
if (module.hot) module.hot.decline();"
`;
